### DISCUSSION TITLE
Restriction being passed from main report to another

### DISCUSSION THREAD

**Discussion by sridharkannan77** [ariacom/Seal-Report#103] [General]
I have a report which displays the following
<img width="946" alt="Screenshot 2024-06-04 215625" src="https://github.com/ariacom/Seal-Report/assets/19453253/505b37cf-34cc-4d01-ac3c-7a476f32859c">

The number on the first column and 1st row is "9". This column is an aggregate column. On clicking of this number, I have to pass the restriction values shown at the top and also the value "SAL" and the value "0 - 28 Lacs" which is displayed above the number "9". These are all paameters that needs to be passed to the second report.

I have created another report in the "reports" folder itself and am using the following script which is written in the cell that displays "9"

@{
    ResultCell cell = Model;

    ReportElement element = cell.Element;
    ReportModel reportModel = element.Model;
    Report report = reportModel.Report;
    
    //Report Execution in the navigation context
    cell.AddNavigationReportNavigation("\\Indiabulls\\SubreportLoginDetails.srex", "View details for " + cell.DisplayValue, "Navigation");
    //cell.AddNavigationReportNavigation("\\SubreportLoginDetails.srex", "View Orders " + cell.ContextRow);
    
}

Now there are 2 questions 
1. How to pass the all the parameters from the main report to this report that is being called. This includes the restrictions shown at the top and also the 2 column values shown above the number "9"
2. What needs to be written in the other report being called?

Any response or any link that does a similar thing will be highly appreciated

**Comment by ariacom** [comment]
I have checked the code, and it looks that restrictions of the report are not propagated to the sub-report.
The code that create the link URL is in ResultCell.cs line 633:
```
                   //Get sub reports links
                   if (Element.PivotPosition != PivotPosition.Data)
                   {
                       foreach (var subreport in Element.MetaColumn.SubReports.Where(i => i.Restrictions.Count > 0))
                       {
                           string subReportRestrictions = "";
                           int index = 1;
                           foreach (var guid in subreport.Restrictions)
                           {
                               var cellValue = SubReportValues.FirstOrDefault(i => i.Element.MetaColumnGUID == guid);
                               if (cellValue != null && !string.IsNullOrEmpty(cellValue.NavigationValue))
                               {
                                   subReportRestrictions += string.Format("&res{0}={1}&val{0}={2}", index, guid, HttpUtility.UrlEncode(cellValue.NavigationValue));
                                   index++;
                               }
                           }
```
Thus, it could be a feature request for a future version of Seal Report to allow (or not) this propagation.

You may try to add restrictions values to the link.Href property (e.g. for a simple subreport: rpa=%25SEALREPOSITORY%25%5cSubReports%5cEmployee+Detail.srex&dis=Buchanan&res1=34215f49-e6a5-45e8-bfb8-d51c3a8f0033&val1=5)
so you may add res2, res3, etc. that comes from your report dynamically in the cell script (Links is the list of link for a ResultCell).
